daisy/workflow.go

```diff
@@ -492,9 +492,7 @@ func (w *Workflow) AddDependency(dependent *Step, dependencies ...*Step) error {
 	return nil
 }
 
-// NewIncludedWorkflow instantiates a new workflow with the same resources as the parent.
-func (w *Workflow) NewIncludedWorkflow() *Workflow {
-	iw := New()
+func (w *Workflow) includeWorkflow(iw *Workflow) {
```

Comment:
To me, the original way looks more natural. Why do we change to the new way?

Reply:
the reason for this change is that the 'parenting' this function does is necessary for included workflows to function properly. if you are building workflows from JSON, the NewIncludedWorkflow function was called (from NewIncludedWorkflowFromFile). But if you've crafted a workflow in code, it wasn't ever called. So essentially you could not create a working IncludeWorkflow step in code. This PR splits the parenting from the workflow creation, so that it can be used in both cases.